tenable_io/client.py

```diff
@@ -46,20 +43,35 @@ def __init__(
         self._endpoint = endpoint
         self._impersonate = impersonate
 
-        self._headers = {
-            u'X-ApiKeys': u'accessKey=%s; secretKey=%s;' % (self._access_key, self._secret_key),
-            u'User-Agent': u'TenableIOSDK Python/%s' % ('.'.join([str(i) for i in sys.version_info][0:3]))
-        }
-
-        if impersonate is not None:
-            self._headers[u'X-Impersonate'] = u'username=%s' % self._impersonate
-
+        self._init_session()
         self._ini_api()
         self._init_helpers()
 
+    def _init_session(self):
+        """
+        Initializes the requests session
+        """
+        retries = Retry(
+                    total=3,
+                    status_forcelist={429}, 
```

Comment:
I don't see any reason why not.  I was keeping it narrowly focused on 429s, but no reason not to expand it.